src/fastoad/io/variable_io.py

```diff
@@ -11,34 +11,56 @@
 #  You should have received a copy of the GNU General Public License
 #  along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
+import logging
 from fnmatch import fnmatchcase
-from os.path import exists, isfile
-from typing import IO, List, Sequence, Union
+from io import IOBase
+from os import PathLike
+from pathlib import Path
+from typing import List, Sequence, Union, IO, Optional
 
 from fastoad.openmdao.variables import VariableList
 from . import IVariableIOFormatter
 from .xml import VariableXmlStandardFormatter
+from .._utils.files import as_path
 from ..exceptions import FastError
 
+_LOGGER = logging.getLogger(__name__)  # Logger for this module
+
 
 class VariableIO:
     """
     Class for reading and writing variable values from/to file.
 
     The file format is defined by the class provided as `formatter` argument.
 
-    :param data_source: the I/O stream, or a file path, used for reading or writing data
+    :param data_source: I/O stream, or file path, used for reading or writing data
     :param formatter: a class that determines the file format to be used. Defaults to a
                       VariableBasicXmlFormatter instance.
     """
 
-    def __init__(self, data_source: Union[str, IO], formatter: IVariableIOFormatter = None):
+    def __init__(
+        self,
+        data_source: Optional[Union[str, PathLike, IO, IOBase]],
+        formatter: IVariableIOFormatter = None,
+    ):
+        if isinstance(data_source, (str, PathLike)):
+            data_source = as_path(data_source)
+
+        #: I/O stream, or file path, used for reading or writing data
         self.data_source = data_source
-        self.formatter: IVariableIOFormatter = (
-            formatter if formatter else VariableXmlStandardFormatter()
-        )
 
-    def read(self, only: List[str] = None, ignore: List[str] = None) -> VariableList:
+        self.formatter = formatter
+
+    @property
+    def formatter(self) -> IVariableIOFormatter:
+        """Class that determines the file format to be used."""
+        return self._formatter
+
+    @formatter.setter
+    def formatter(self, formatter: IVariableIOFormatter):
+        self._formatter = formatter if formatter else VariableXmlStandardFormatter()
```

Comment:
Any other reason behind this change other than making it cleaner ?

Reply:
Yes. It allows to get back to the default formatter when setting the property to None.
The DataFile class has been reworked and can now set the formatter after having instantiated its internal VariableIO object.